### ISSUE TITLE
Unable to set fixed=true through modifier field

### ISSUE THREAD

**Issue by OpenModelica-TracImporter** [OpenModelica/OpenModelica#5105]
Original ticket https://trac.openmodelica.org/OpenModelica/ticket/5105

**Comment by adeas31** [comment]
@ceraolo I believe this issue is not valid anymore. However, I can't test it properly since the example model is very old. Can you please test this. Post a new test model if the problem is still valid.

**Comment by ceraolo** [comment]
Before posting I tested this yesterday. 
I'd just to download MSL 4.0.0 and load 3.2.3. 

However, this is converted RL3:
[RL3_4.mo.txt](https://github.com/OpenModelica/OpenModelica/files/15014913/RL3_4.mo.txt)

**Comment by casella** [comment]
@adeas31, the original model used MSL 3.2.3 but had a bogus uses annotation. [RL3.mo.txt](https://github.com/OpenModelica/OpenModelica/files/15017197/RL3.mo.txt) fixes that.

What is strange is that I cannot set the inductor's current start and and fixed attributes through the GUI for the existing components in the diagram, but if I drag and drop a new inductor and double-click on it, then the checkbox for fixed and the input field for start are present. Why is that happening?

**Comment by adeas31** [comment]
@casella are you sure you tested with `Modelica.Electrical.MultiPhase.Basic.Inductor` (for MSL 4.0.0 it is `Modelica.Electrical.Polyphase.Basic.Inductor`) and not with `Modelica.Electrical.Analog.Basic.Inductor`?

**Comment by casella** [comment]
> @casella are you sure you tested with `Modelica.Electrical.MultiPhase.Basic.Inductor` (for MSL 4.0.0 it is `Modelica.Electrical.Polyphase.Basic.Inductor`) and not with `Modelica.Electrical.Analog.Basic.Inductor`?

Of course you're right, sorry 😅

**Comment by adeas31** [comment]
> > @casella are you sure you tested with `Modelica.Electrical.MultiPhase.Basic.Inductor` (for MSL 4.0.0 it is `Modelica.Electrical.Polyphase.Basic.Inductor`) and not with `Modelica.Electrical.Analog.Basic.Inductor`?
> 
> Of course you're right, sorry 😅

Then I don't understand what is wrong. Maybe I am still missing something. @ceraolo can you please check this again and report what is wrong.

**Comment by casella** [comment]
@ceraolo your test model uses `Modelica.Electrical.MultiPhase.Basic.Inductor,` which is implemented internally by instantiating an array of N basic `inductor` components. Hence, the fixed attribute must be applied to a nested modifier, e.g.
```
Modelica.Electrical.MultiPhase.Basic.Inductor inductor1(L = fill(0.01, 3), inductor(i(each start = 1, each fixed = true)));
```
This feature is currently not yet supported by the OMEdit GUI, but is planned to be added this year, see #2891. So, this is basically a duplicate of #2891. Do you want to keep it open explicitly as a reminder?

**Comment by ceraolo** [comment]
I think it is better to keep it open.I periodically check all my tickets to see whether they are solved in the meanwhile and in case close them.Il 22 Apr 2024 14:01, Francesco Casella ***@***.***> ha scritto:
@ceraolo your test model uses Modelica.Electrical.MultiPhase.Basic.Inductor, which is implemented internally by instantiating an array of N basic inductor components. Hence, the fixed attribute must be applied to a nested modifier, e.g.
Modelica.Electrical.MultiPhase.Basic.Inductor inductor1(L = fill(0.01, 3), inductor(i(each start = 1, each fixed = true)));

This feature is currently not yet supported by the OMEdit GUI, but is planned to be added this year, see #2891. So, this is basically a duplicate of #2891. Do you want to keep it open explicitly as a reminder?

**Comment by casella** [comment]
@ceraolo, setting the initial currents should now be easier in MSL 4.1.0, see modelica/ModelicaStandardLibrary#4488.

You can already try this out by installing Modelica 4.1.0 pre-release build beta.om. We will release MSL 4.1.0 early May, then it will be pre-installed together with MSL 4.0.0.

**Comment by adeas31** [comment]
Issue #2891 is resolved, and hierarchical editing is already available.

@ceraolo, feel free to reopen the issue if the problem persists.

**Comment by ceraolo** [comment]
So, @adeas31 it is now possible to remove the warnings mentioned in 
https://github.com/OpenModelica/OpenModelica/issues/5105#issuecomment-2067185475
through the GUI without resorting to MSL 4.1.0 (as is possible dwith Dymola?
I could not find a way using today's nigthly (1.26.0-dev107)
Thanks 

@casella nice knowing that MSL 4.1.0 enhances setting initial currents in inductors. Probably they've started to implement something I proposed them a few months ago.

**Comment by adeas31** [comment]
> So, [@adeas31](https://github.com/adeas31) it is now possible to remove the warnings mentioned in [#5105 (comment)](https://github.com/OpenModelica/OpenModelica/issues/5105#issuecomment-2067185475) through the GUI without resorting to MSL 4.1.0 (as is possible dwith Dymola? I could not find a way using today's nigthly (1.26.0-dev107) Thanks

Yeah this is not yet possible. I will try to fix this asap. Should be an easy fix. I guess the add modifier regular expression fails.

**Comment by adeas31** [comment]
@ceraolo try #13901 with the tomorrow's nightly build.
Note that you need to set `i(each start=0, each fixed=true)` instead of `i(each start=0, fixed=true)` in order to get the model working.

**Comment by casella** [comment]
> [@casella](https://github.com/casella) nice knowing that MSL 4.1.0 enhances setting initial currents in inductors. Probably **they**'ve started to implement something I proposed them a few months ago.

Sure, see the discussion in modelica/ModelicaStandardLibrary#4457. This was done with PR modelica/ModelicaStandardLibrary#4488 in October 2024.

BTW, by "they" you mean "you", I'm actually leading the development of MSL at the moment 😄

**Comment by casella** [comment]
FYI, MSL 4.1.0 will be voted by the MA until May 18, if all goes well it will be released on that date. It will then immediately become available in all versions of OpenModelica through the Package Manager.

**Comment by ceraolo** [comment]
> BTW, by "they" you mean "you", I'm actually leading the development of MSL at the moment 😄

AHA! so "you" yourself mentioned that MSL 4.1.0 had done an improvement as if the programmers had got an insight from the heavens, while, in reality, whey just followed a recommendation by me.
;-)

**Comment by ceraolo** [comment]
> [@ceraolo](https://github.com/ceraolo) try [#13901](https://github.com/OpenModelica/OpenModelica/pull/13901) with the tomorrow's nightly build. 

it seems that 1.26.0 dev-113 still does not include this correction: 
![Image](https://github.com/user-attachments/assets/85850f45-7a2c-41a8-ae31-c51f95bc5dad)

I will try again the next days.

**Comment by adeas31** [comment]
The error message looks correct to me so the change is included. Seems you are setting `each start=0, each fixed=true` which of course gives the error. You need to set `i(each start=0, each fixed=true)`.

**Comment by ceraolo** [comment]
bleah! 
Sorry for this mistake. I confirm everything is fine.

Thanks
